Feature/password history

Abdulrahman7533 opened the pull request:
**1) Student Name:**
Abdurahman Saleh M Alghaith

**2) Task Description:**
Implement Password History (prevent reuse of current and last 2 passwords)

**3) What I Did:**

1. Added PasswordHistory model (stores encoded password + timestamp).
2. Implemented PasswordHistoryValidator using check_password() to compare against the current and last 2 hashes.
3. History storage is handled via signals:
- post_save  on user creation to capture the initial password.
- pre_save on user update to record the previous (old) encoded password before changes.
- Both signals prune older entries to keep only the latest 2.
- Updated INSTALLED_APPS entry to home.apps.HomeConfig  to ensure signals are loaded on startup.


4) Screenshots:

Case 1: Change password to the same as the current one.  
Result: Rejected with error: “You cannot reuse your current password.”
<img width="222" height="297" alt="image" src="https://github.com/user-attachments/assets/20d4896b-ae87-4e7e-8311-eafeacf451a2" />

 
Case 2: Change password to one of the last 2 used 
Result: Rejected with error: “You cannot reuse one of your recent passwords.”
 
<img width="224" height="299" alt="image" src="https://github.com/user-attachments/assets/ae5a17d5-b068-4de9-874a-8c2071dc1e96" />

drkae456 reviewed (changes requested):
Hi , please dont create extra migration files, edit the initial migration file and models.py to change database

Abdulrahman7533 commented:
Hi, I reset models.py and 0001_initial.py to match the upstream. Then I added the PasswordHistory model again in models.py, placed its CreateModel block inside 0001_initial.py, and removed the extra migration file that was created before.

drkae456 reviewed (approved):
Good Job! Approved
